internal/types/path.go

```diff
@@ -263,6 +265,14 @@ func insertByPath(doc *Document, path Path) error {
 						),
 					)
 				}
+
+				// If path needs to be reserved in the middle of the array, we should fill the gap with Null
```

Comment:
@chilagrow PTAL, it looks similat to your `$inc` task!

Reply:
This is great!